lib/editor.coffee

```diff
@@ -43,34 +44,41 @@ textEditor = ($item, item, option={}) ->
       return false
 
     # provides automatic new paragraphs on enter and concatenation on backspace
-    if item.type is 'paragraph'
+    if item.type is 'paragraph' or 'markdown'
       sel = getSelectionPos($textarea) # position of caret or selected text coords
 
       if e.which is $.ui.keyCode.BACKSPACE and sel.start is 0 and sel.start is sel.end
         $previous = $item.prev()
         previous = itemz.getItem $previous
-        return false unless previous.type is 'paragraph'
+        return false unless previous.type is item.type
         caret = previous[option.field||'text'].length
         suffix = $textarea.val()
         $textarea.val('') # Need current text area to be empty. Item then gets deleted.
         textEditor $previous, previous, {caret, suffix}
         return false
 
       if e.which is $.ui.keyCode.ENTER
+        # console.log "Type: #{item.type}, enterCount: #{enterCount}"
         return false unless sel
-        $page = $item.parents('.page')
-        text = $textarea.val()
-        prefix = text.substring 0, sel.start
-        suffix = text.substring(sel.end)
-        if prefix is ''
-          $textarea.val(suffix)
-          $textarea.focusout()
-          spawnEditor($page, $item.prev(), prefix, true)
```

Comment:
The `true` in this call looks to be redundant, `spawnEditor` only has 3 parameters, and contains no code to retrieve any extra arguments, so it has been removed.